spark-indexer/src/main/java/org/uniprot/store/spark/indexer/uniparc/mapper/UniParcEntryJoin.java

```diff
@@ -0,0 +1,117 @@
+                    getMappedSourcesWithProteomes(
+                            result.getUniParcCrossReferences(), join.getSequenceSources());
+
+            List<UniParcCrossReference> mappedXRefs =
+                    result.getUniParcCrossReferences().stream()
+                            .map(xref -> mapTaxonomy(xref, mappedTaxons))
+                            .map(xref -> mapSequenceSource(xref, sourceMap))
+                            .collect(Collectors.toList());
+            builder.uniParcCrossReferencesSet(mappedXRefs);
+            result = builder.build();
+        }
+        return result;
+    }
+
+    private UniParcCrossReference mapSequenceSource(
+            UniParcCrossReference xref, Map<String, Set<String>> sourceMap) {
+        Set<String> sources = sourceMap.get(xref.getId());
+        if (Utils.notNullNotEmpty(sources)) {
+            UniParcCrossReferenceBuilder xrefBuilder = UniParcCrossReferenceBuilder.from(xref);
+            xrefBuilder.propertiesAdd(new Property(PROPERTY_SOURCES, String.join(",", sources)));
+            xref = xrefBuilder.build();
+        }
+        return xref;
+    }
+
+    private Map<String, Set<String>> getMappedSourcesWithProteomes(
+            List<UniParcCrossReference> xrefs, Map<String, Set<String>> sourceMap) {
+        Map<String, Set<String>> result = new HashMap<>();
+        if (sourceMap != null) {
+            Map<String, UniParcCrossReference> mappedCrossReference =
+                    getMappedCrossReference(xrefs);
+            result =
+                    sourceMap.entrySet().stream()
+                            .collect(
+                                    Collectors.toMap(
+                                            Map.Entry::getKey,
+                                            source -> addProteome(source, mappedCrossReference)));
+        }
+        return result;
+    }
+
+    @NotNull
+    private static Map<String, UniParcCrossReference> getMappedCrossReference(
+            List<UniParcCrossReference> xrefs) {
+        return xrefs.stream()
+                .collect(
+                        Collectors.toMap(
+                                UniParcCrossReference::getId,
+                                xref -> xref,
+                                (xref1, xref2) -> xref1.isActive() ? xref1 : xref2));
+    }
+
+    @NotNull
+    private static Set<String> addProteome(
+            Map.Entry<String, Set<String>> e,
+            Map<String, UniParcCrossReference> mappedCrossReference) {
+        return e.getValue().stream()
+                .map(mappedCrossReference::get)
+                .filter(xref -> xref != null && xref.getProteomeId() != null)
+                .map(UniParcEntryJoin::getAccessionWithProteome)
+                .collect(Collectors.toSet());
+    }
+
+    private static String getAccessionWithProteome(UniParcCrossReference xref) {
+        return xref.getId() + ":" + xref.getProteomeId() + ":" + xref.getComponent();
+    }
+}
```

Comment:
Is xref.getId() always an accession? I don't think so. The method name doesn't include component information.
Could you please rename getAccessionWithProteome?